docs/proposals/30-apiserver-proxy.md

```diff
@@ -0,0 +1,268 @@
+    
+    
+## Custom Header Specification
+
+The new custom header will be structured as follows:
+
+```
+X-Gardener-API-Route: <routing-information>
+```
+
+The routing information will be a concatenated string, joined by pipes, containing the following fields:
+
+- `direction`: The direction of traffic.
+- `port`: The destination port.
+- `subset`: Istio subset/version.
+- `destination`: The destination.
+
+Putting this to practice should result in a header formatted like this:
+
+`X-Gardener-API-Route: outbound|8134||kube-apiserver.foobar.svc.cluster.local`
+
+During the transition period, the system should be able to handle both the new custom header and the existing routing method. This can be controlled via the feature gate.
+
+If the header is missing, malformed, or fails verification, the error should be logged with appropriate details for debugging and return an HTTP 400 (Bad Request) status code. It should never fall back to the old routing method to ensure security.
+
+## HTTP CONNECT Implementation
+
+The `apiserver-proxy` will be reconfigured to use HTTP CONNECT for establishing a secure tunnel to the `kube-apiserver`. This involves:
+
+1. Initiating an HTTP CONNECT request to the new port/path on the `istio-ingressgateway`
+2. Including the new custom header in the CONNECT request
+3. Creating a secure connection over the established tunnel once the CONNECT request is accepted
+4. Forwarding the original API server request through this secure tunnel
+
+### Technical Implementation Details
+
+The `apiserver-proxy` will be required to be updated to use HTTP CONNECT. This should involve modifying its configuration and potentially its code. One example of how the configuration might look like after reusing its Envoy tunneling config:
+
+```yaml
+...
+- domains: 
+  - api.*
+  name: gardener-api-route
+  routes:
+  - match:
+      connect_matcher: {}
+    route:
+     cluster_header: X-Gardener-API-Route
+     upgrade_configs:
+     - connect_config: {}
+       upgrade_type: CONNECT
+...
+```
+
+This solution would be reusing existing Envoy filtering on the `apiserver-proxy` pod and simply exchange its Proxy Protocol configuration for the new proposed header.
+
+### Istio IngressGateway Configuration
+
+Similarly, it would be required to configure the Istio IngressGateway to accept HTTP CONNECT requests and route them appropriately. This could e.g. be achieved utilizing Istio Custom Resources (Gateway, VirtualService):
+
+```yaml
+apiVersion: networking.istio.io/v1alpha3
+kind: Gateway
+metadata:
+  name: api-gateway
+  namespace: istio-system
+spec:
+  selector:
+    istio: ingressgateway
+  servers:
+    ...
+---
+apiVersion: networking.istio.io/v1alpha3
+kind: VirtualService
+metadata:
+  name: api-routes
+  namespace: istio-system
+spec:
+  ...
+  gateways:
+  - api-gateway
+  http:
+  - match:
+    - method:
+        exact: CONNECT
+      headers:
+        X-Gardener-API-Route:
+          regex: ".*"
```

Comment:
We probably do not want to do this, but instead use an external authorization service just like with the VPN. It should not be possible to reach random kubernetes services in the seed using this approach. After all, we want to close security vulnerabilities and not open new ones ;-) .
Maybe you just link this to the next section and make it explicit that not just any random header value will be accepted.